File: backend/src/MicroclimateIotSystem.Domain/Entities/AlertRule.cs
using MicroclimateIotSystem.Domain.Abstractions;
using MicroclimateIotSystem.Domain.Enums;

namespace MicroclimateIotSystem.Domain.Entities;

public class AlertRule : BaseEntity
{
    public string Name { get; set; }
    public string TelemetryKey { get; set; }
    public AlertRuleOperator Operator { get; set; }
    public double ThresholdValue { get; set; }
    public bool IsActive { get; set; }
    public int? RoomId { get; set; }
    public int? DeviceId { get; set; }
    
    public Room? Room { get; set; }
    public Device? Device { get; set; }
}

File: frontend/src/app/models/alert-rule.ts
export enum AlertRuleOperator {
  GreaterThan = 'GreaterThan',
  GreaterThanOrEqualTo = 'GreaterThanOrEqualTo',
  LessThan = 'LessThan',
  LessThanOrEqualTo = 'LessThanOrEqualTo',
}

export const ALERT_RULE_OPERATOR_LABELS: Record<AlertRuleOperator, string> = {
  GreaterThan: '>',
  GreaterThanOrEqualTo: '≥',
  LessThan: '<',
  LessThanOrEqualTo: '≤',
};

export interface AlertRule {
  id: number;
  name: string;
  telemetryKey: string;
  operator: AlertRuleOperator;
  thresholdValue: number;
  isActive: boolean;
  roomId: number | null;
  roomName: string | null;
  deviceId: number | null;
  deviceName: string | null;
}

export interface CreateAlertRuleRequest {
  name: string;
  telemetryKey: string;
  operator: AlertRuleOperator;
  thresholdValue: number;
  roomId: number | null;
  deviceId: number | null;
}